inc/datahandlers/user.php

```diff
@@ -1855,10 +1855,13 @@ public function verify_signature()
 			$parsed_sig = $this->data['signature'];
 		}
 
-		$parsed_sig = preg_replace("#\s#", "", $parsed_sig);
-		$sig_length = my_strlen($parsed_sig);
+		if($mybb->settings['siglength'] > 0)
+		{
+			$parsed_sig = preg_replace("#\s#", "", $parsed_sig);
+			$sig_length = my_strlen($parsed_sig);
+		}
 
-		if($sig_length > $mybb->settings['siglength'])
+		if($sig_length > $mybb->settings['siglength'] && $mybb->settings['siglength'] > 0)
```

Comment:
Sorry to be a bother but I think `$mybb->settings['siglength'] > 0` should be the first check in case `$sig_length` isn't set.

@dvz Am I on the right track or I'm just saying nonsense and we could get this merged?

Reply:
It looks like this block can be moved inside the one above to solve this issue and simplify the code.